### PR TITLE
Story/ccls 2181 app submission summary screen

### PR DESCRIPTION
**Pull request by PhilDigitalJustice**
Includes: CCLS-2303 (caab-ui: refactor application summaries and section data objects)

This task involves creating a final case submission summary screen in the CAAB-UI, which is displayed after all validation is complete. The screen will provide a detailed summary of various sections of the application, including client details, general details, provider details, proceedings and costs, opponents and other parties, means assessments, and merits assessment. The screen will also have a "Print" button (covered in another story), a "Next" button that navigates to the sign declaration screen, and a "Back" button that returns the user to the create application section screen.

As a user, 
I want to view a comprehensive summary of my application before submission, 
so that I can review all entered information for accuracy and completeness.

Design:

Client Details:

Basic Client Details
Contact Details
Address Details
Equal Opportunities Monitoring

General Details:

Correspondence Address
Provider Details

Proceedings and Costs:

Each Proceeding's Details
Each Proceeding's Scope Limitations

Opponents and Other Parties:
Individuals
Companies
Means Assessments
Merits Assessment

Technical Implementation:

Create Submission Summary Screen:
Develop the screen to display the application summary after validation is complete.
Use the provided screenshot as a reference for the layout and design.

Display Summary Information:

Fetch and display all relevant data sections:

Client Details (basic, contact, address, equal opportunities monitoring)
General Details (correspondence address)
Provider Details
Proceedings and Costs (details, scope limitations)
Opponents and Other Parties (individuals, companies)
Means Assessments
Merits Assessment

Navigation Buttons:

Implement the "Print" button (functionality covered in another story).
Implement the "Next" button to navigate to the sign declaration screen.
Implement the "Back" button to return to the create application section screen.

Data Fetching:

Ensure the necessary data is fetched and displayed correctly.
Handle any data loading states and errors appropriately.
Use asynchronous methods where possible 

Acceptance Criteria:

The submission summary screen displays all required sections accurately.
Users can review all details of their application in a clear and organised format.
The "Print" button is present (functionality to be implemented in another story).
The "Next" button navigates to the sign declaration screen.
The "Back" button returns the user to the create application section screen.
Comprehensive testing is conducted to ensure all information is displayed correctly and navigation works as intended.

### CODE REVIEW

**Review by farrell-m — CHANGES_REQUESTED** [review]
Nice job 👍  Just a few small comments

File: build.gradle [hunk 1]
```diff
@@ -38,7 +38,7 @@ dependencies {
 
 	implementation 'uk.gov.laa.ccms.data:data-api:0.0.12'
 
-	implementation 'uk.gov.laa.ccms.soa.gateway:soa-gateway-api:0.0.25'
+	implementation 'uk.gov.laa.ccms.soa.gateway:soa-gateway-api:0.0.26-9346235-SNAPSHOT'
```
[the remark this answers is not in the document]
Comment (PhilDigitalJustice) [reply build.gradle:41]: Will do this using the automated pr after this is reviewed

File: src/main/java/uk/gov/laa/ccms/caab/service/LookupService.java [hunk 2]
```diff
@@ -398,4 +415,205 @@ public Mono<CommonLookupDetail> getCommonValues(final String type) {
...
+  }
+
+  /**
+   * Adds common lookups to the model and retrieves a list of lookup details.
+   *
+   * @param lookups The list of pairs containing lookup keys and Mono optional values.
+   * @param model The model to which lookups will be added.
+   * @return A Mono containing a list of lookup details.
+   */
+  public Mono<List<CommonLookupValueDetail>> addCommonLookupsToModel(
```
[the remark this answers is not in the document]
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/service/LookupService.java:476]: Yes you are correct, i will also adjust the other common abstract method that uses this.

File: src/test/java/uk/gov/laa/ccms/caab/controller/application/section/ApplicationSubmissionControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            .sessionAttr(USER_DETAILS, userDetail)
+            .sessionAttr(ACTIVE_CASE, activeCase))
+        .andExpect(status().isOk())
+        .andExpect(view().name("application/sections/application-summary-complete"))
+        .andExpect(model().attributeExists("submissionSummary"));
+
+    Mockito.verify(applicationService, Mockito.times(1)).getApplication(any());
+    Mockito.verify(assessmentService, Mockito.times(1)).getAssessments(any(), any(), any());
+    Mockito.verify(clientService, Mockito.times(1)).getClient(any(), any(), any());
+    Mockito.verify(lookupService, Mockito.times(2)).getAssessmentSummaryAttributes(any());
```
[the remark this answers is not in the document]
Comment (PhilDigitalJustice) [reply src/test/java/uk/gov/laa/ccms/caab/controller/application/section/ApplicationSubmissionControllerTest.java:171]: Amended

File: src/test/java/uk/gov/laa/ccms/caab/controller/application/client/ClientSummaryControllerTest.java [hunk 4]
```diff
@@ -113,82 +118,80 @@ public void setup() {
...
 
-    verify(lookupService, atLeastOnce()).getCommonValue(eq(COMMON_VALUE_CONTACT_TITLE),  any());
-    verify(lookupService, atLeastOnce()).getCountry(any());
-    verify(lookupService, atLeastOnce()).getCommonValue(eq(COMMON_VALUE_GENDER), any());
-    verify(lookupService, atLeastOnce()).getCommonValue(eq(COMMON_VALUE_MARITAL_STATUS), any());
-    verify(lookupService, atLeastOnce()).getCommonValue(eq(COMMON_VALUE_ETHNIC_ORIGIN),any());
-    verify(lookupService, atLeastOnce()).getCommonValue(eq(COMMON_VALUE_DISABILITY),any());
-    verify(lookupService, atLeastOnce()).getCommonValue(eq(COMMON_VALUE_CORRESPONDENCE_METHOD), any());
+    verify(lookupService, atLeastOnce()).getClientLookups(any());
+    verify(lookupService, atLeastOnce()).addCommonLookupsToModel(anyList(), any(Model.class));
```
[the remark this answers is not in the document]
Comment (PhilDigitalJustice) [reply src/test/java/uk/gov/laa/ccms/caab/controller/application/client/ClientSummaryControllerTest.java:151]: Amended

**Pull request merged**